File: competition-serv-model/src/main/java/compman/compsrv/model/events/payload/BracketsGeneratedPayload.java
package compman.compsrv.model.events.payload;

import compman.compsrv.annotations.EventPayload;
import compman.compsrv.model.Payload;
import compman.compsrv.model.dto.brackets.StageDescriptorDTO;
import compman.compsrv.model.events.EventType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serializable;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EventPayload(type = EventType.BRACKETS_GENERATED)
@Accessors(chain = true)
public class BracketsGeneratedPayload implements Serializable, Payload {
    private StageDescriptorDTO[] stages;
}

File: competition-serv-service/src/main/java/compman/compsrv/model/es/events/payload/BracketsGeneratedPayload.kt
package compman.compsrv.model.es.events.payload

import compman.compsrv.model.brackets.BracketType
import compman.compsrv.model.competition.FightDescription

data class BracketsGeneratedPayload(val fights: Array<FightDescription>, val bracketType: BracketType) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as BracketsGeneratedPayload

        if (!fights.contentEquals(other.fights)) return false
        if (bracketType != other.bracketType) return false

        return true
    }

    override fun hashCode(): Int {
        var result = fights.contentHashCode()
        result = 31 * result + bracketType.hashCode()
        return result
    }
}